### DISCUSSION TITLE
Default maximum parameters bounds (pmax) of the response functions

### DISCUSSION THREAD

**Discussion by martinvonk** [pastas/pastas#612] [Ideas]
**Context**
I have difficulty optimizing models with a Gamma response for precipitation-excess and an Exponential response for surface water levels. When adding the surface water level to the model the response of the precipitation excess sometimes get's too long and fits a trend. 

**Issue**
I think this can be (partially) related to the original parameter bounds chosen for the Exponential and Gamma response function. If the parameters are at the default outer bound (`pmax`), the tmax of the Exponential is ~7,000 days (19 years) while the tmax of the Gamma could reach ~1,340,000 days (3669 years). I think we could chose more sensible bounds for these two response functions, such that the maximum tmax of both response functions are in the same range. 

**Discussion Topic**
Since I was working on this I made a small script to check the (maximum) tmax of all response functions given the default outer parameter bound `pmax`. This way we can have a more general discussion on what the "best" parameter bounds could be for the response functions.

```python
import pastas as ps

rfuncs = {n: rf for n, rf in ps.rfunc.__dict__.items() if n in ps.rfunc.__all__ and n != "HantushWellModel"}

for rfname in rfuncs:
    rfunc = rfuncs[rfname]()  # create response function instance
    ipmax = rfunc.get_init_parameters(name=rfname)["pmax"].values  # get maximum parameter bound (pmax)
    # ipmax = [ipmax[0], 100.0, ipmax[2]]
    tmax = rfunc.get_tmax(p=ipmax)  # calculate tmax given pmax
    print(f"{rfname}: {tmax=:0.2f}d, p={ipmax}")
```

```output
Gamma tmax=1337702.64 [d]
Exponential tmax=6907.76 [d]
Hantush tmax=151286.01 [d]
Polder tmax=12825.53 [d]
One tmax=0.00 [d] # bounded at 0.0
FourParam tmax=9999.00 [d] # bounded at 9999.0
DoubleExponential tmax=34538.78 [d]
Edelman tmax=1.27 [d]
Kraijenhoff tmax=690775.53 [d]
Spline tmax=1024.00 [d]

pastas\rfunc.py:1240: RuntimeWarning: invalid value encountered in cast
  y[1:] = y[0] + np.cumsum(1 / 6 * (func[:-1] + 4 * func_half + func[1:])) # error from FourParam
```

**Comment by martinvonk** [comment]
My proposal for the Gamma and Exponential response initial parameter bounds.

|  | Current |  | Proposal |  |
|---|---|---|---|---|
|  | pmax | tmax [d] | pmax | tmax [d] |
| Gamma | a=1e4; n=100.0 | 13337703 | a=1e4; n=5.0 | 130622 |
| Exponential | a=1e3 | 6908 | a=1e4 | 69078 |

**Reply by OnnoEbbens** [reply]
This seems like a proper solution for all the times that the tmax of my response function explodes. We could also think about setting the parameters of the response function based on the length of the oseries. If you have 3 years of measurements in your oseries you don't want a response function of more than 3 years.

**Comment by martinvonk** [comment]
Discussion moved to #971
